modules/runner/src/main/java/org/apache/ignite/internal/configuration/ServiceLoaderModulesProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.configuration;
+
+import java.util.ServiceLoader;
+import java.util.stream.Stream;
+import org.apache.ignite.configuration.validation.ConfigurationModule;
+
+/**
+ * TODO: javadoc.
+ */
+public class ServiceLoaderModulesProvider implements ConfigurationModulesProvider {
```

Comment:
I'm not sure that interface/implementation is really required here, do we expect other implementations?

Reply:
I added an interface because I was expecting that an instance of it would be injected (using dependency injection) to whatever place where modules should be obtained. But it turned out that in `IgniteImpl` all the classes are instantiated manually, no injection is used.

How do you think, will this mechanism be used in other places? If not, then the interface does not seem to be needed, I will remove it.

Reply:
IgniteImpl and tests are the only places that will use it. And, as far as I know, having an interface is not mandatory for injection. And we don't have dependency injection in the first place, so I would simplify this part.

Reply:
Removed the interface as requested